content/hardware/07.opta/opta-family/opta/tutorials/17.opta-mqtt/content.md

```diff
@@ -0,0 +1,555 @@
+  while (WiFi.begin(SECRET_SSID, SECRET_PASS) != WL_CONNECTED) { // Keep attempting connection until connected
+    Serial.print(".");  // Print a dot on serial to indicate connection attempt
+    delay(5000);        // Wait 5 seconds between connection attempts
+  }
+
+  mqttClient.setUsernamePassword(MQTT_USER, MQTT_PASSWORD);  // Set the username and password for authentication with the MQTT broker
+  
+  // Connect to MQTT broker using provided credentials
+  if (!mqttClient.connect(MQTT_BROKER, MQTT_PORT)) {  // If MQTT connection fails
+    while (1);  // Halt program execution in an infinite loop
+  }
+
+}
+```
+
+The **loop()** function sends a test message, in this case "hello", to the MQTT broker every 10 seconds.
+
+```cpp
+void loop() {
+    mqttClient.poll();  // Check for incoming messages from MQTT broker
+    
+    // Begin a new MQTT message on the specified topic
+    mqttClient.beginMessage(MQTT_TOPIC);
+    
+    // Add message content (in this case, "hello" followed by a count)
+    mqttClient.print("hello");
+
+    // End the MQTT message and send it to the broker
+    mqttClient.endMessage();
+    
+    delay(10000);  // Wait for 10 seconds before sending the next message
+}
+```
+
+### Connection Configuration
+
+
+In the attached sketch to this tutorial, you will find a support file called **arduino_secrets.h**, which contains the definitions that will allow you to configure the connection to the newly created MQTT broker.
+
+![Sketch for credentials CloudMQTT](assets/opta-mqtt-tutorial-0015.png)
+
+In this file, we can replace the data according to the information generated on CloudMQTT:
+
+```cpp
+#define SECRET_SSID "Your Network"               // WiFi network SSID (name)
+#define SECRET_PASS "Your Network password"      // WiFi network password
+#define MQTT_BROKER "hairdresser.cloudmqtt.com"  // MQTT broker address
+#define MQTT_PORT 18486                           // MQTT broker port
+#define MQTT_TOPIC "arduino/topic"                // MQTT topic to publish messages to
+#define MQTT_USER "rygfvrup"                      // MQTT broker username (if required)
+#define MQTT_PASSWORD "xxxxxxx"                   // MQTT broker password (if required)
+```
+
+**MQTT_TOPIC ?** A "topic" is like a communication channel on which devices can send and receive messages using the MQTT protocol. It's akin to a topic of conversation: when a device publishes a message on a specific topic, other devices interested in that topic can receive and read the message. It's a way to organize and address messages in an IoT network, allowing devices to communicate in a targeted and efficient manner. For example, you might have a topic called "temperature" on which a sensor publishes temperature data and a control device receives them to regulate an air conditioning system. In this tutorial, we use the topic **arduino/topic**, which is not transcendental and can be customized as you prefer.
```

Comment:
@mirkopac what does the `?` mean in the beginning of this section? Is it supposed to work like a `:`?

Reply:
@mirkopac please see my comment here 🙏🏼